infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/query/parsing/impl/InfinispanRemoteBasedQueryParserService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.query.parsing.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+import java.util.Map;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.hql.QueryParser;
+import org.hibernate.hql.ast.spi.EntityNamesResolver;
+import org.hibernate.ogm.query.spi.BaseQueryParserService;
+import org.hibernate.ogm.query.spi.QueryParserService;
+import org.hibernate.ogm.query.spi.QueryParsingResult;
+import org.hibernate.ogm.service.impl.SessionFactoryEntityNamesResolver;
+import org.hibernate.ogm.util.impl.Log;
+import org.hibernate.ogm.util.impl.LoggerFactory;
+
+/**
+ * A {@link QueryParserService} implementation which creates Infinispan server queries in form of {@link String}s.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class InfinispanRemoteBasedQueryParserService extends BaseQueryParserService {
```

Comment:
I'm agree with you, but I just applied what was done with MongoDB and Neo4j, where the similar classes are named MongoDBBasedQueryParserService and InfinispanRemoteBasedQueryParserService.

Reply:
If you want if I change it, I think we have change MongoDBBasedQueryParserService and InfinispanRemoteBasedQueryParserService too.

Reply:
OK, I didn't see that MongoDB already does that, let's keep it with the Base then. We can change both later. This one got my attention because the name is longer than the MongoDB one